webrender/src/prim_store.rs

```diff
@@ -1322,6 +1322,14 @@ impl PrimitiveStore {
             .collect();
 
         if clips.is_empty() {
+            // If this item is in the root coordinate system, then
+            // we know that the local_clip_rect in the clip node
+            // will take care of applying this clip, so no need
+            // for a mask.
+            if prim_coordinate_system_id == CoordinateSystemId::root() {
```

Comment:
I don't fully understand why this is needed. Wouldn't the same apply to any axis-aligned coordinate system, which should already be handled?

Reply:
It doesn't seem to handle it - @mrobinson thinks there might be a deeper issue that he is looking into. If / when that's resolved we could hopefully remove this.